core/src/main/java/com/orientechnologies/orient/core/index/OIndexManagerShared.java

```diff
@@ -129,7 +129,7 @@ public OIndex createIndex(
             || indexDefinition.getClassName() == null
             || indexDefinition.getFields() == null
             || indexDefinition.getFields().isEmpty();
-    if (manualIndexesAreUsed) {
+    if (manualIndexesAreUsed && !"LUCENE_CROSS_CLASS".equals(algorithm)) {
```

Comment:
I would not include the specific check for "LUCENE_CROSS_CLASS" and just leave the default check.

Reply:
It looks like the cross class index isn't really a manual index, it just collides with the manual index logic as it's created as a special case index. This was just replicating the logic in `OCreateIndexStatement.execute`, which special cases the index.
It should probably be modelled as a distinct type of index (i.e. maybe a virtual index) that has no key fields, but since the existing hack was in place I replicated it rather than extend the index SPI.

As you say, we can work around that by enabling manual indexes, but it's a bit messy either way.